### PR TITLE
HPCC-21258 Track CallerId through HPCC components

### PR DESCRIPTION
**Pull request by afishbeck**
Also support "Global-Id", "Caller-Id" versions of the id tracking
HTTP headers.  Also remember the form of HTTP header used and use
the same in SOAPCALL and HTTPCALL.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: rtl/include/eclhelper.hpp [hunk 1]
```diff
@@ -44,8 +44,8 @@ typedef unsigned short UChar;
 
 //Should be incremented whenever the virtuals in the context or a helper are changed, so
 //that a work unit can't be rerun.  Try as hard as possible to retain compatibility.
-#define ACTIVITY_INTERFACE_VERSION      650
-#define MIN_ACTIVITY_INTERFACE_VERSION  650             //minimum value that is compatible with current interface
+#define ACTIVITY_INTERFACE_VERSION      651
```
Comment (afishbeck) [inline rtl/include/eclhelper.hpp]: @ghalliday this seems like overkill when the only way ECL can access the new function is through a new plugin method.  But a workunit that uses the new plugin feature will not be backward compatible with 7.0.x.

Is there a better way?

Comment (richardkchapman) [reply rtl/include/eclhelper.hpp]: I don't think we change the interface version for cases where new functions have been added and only new code uses it.

**Comment by afishbeck** [comment]
@richardkchapman removed eclhelper version number increment.

**Comment by richardkchapman** [comment]
@wangkx Please review

File: thorlcr/graph/thgraph.cpp [hunk 2]
```diff
@@ -2577,6 +2582,10 @@ class CThorContextLogger : implements IContextLogger, public CSimpleInterface
     {
         return localId.str();
     }
+    virtual const char *queryCallerId() const override
+    {
+        return localId.str();
```
Comment (wangkx) [inline thorlcr/graph/thgraph.cpp]: Should be callerId?

**Review by wangkx — COMMENTED** [review]
@afishbeck one question.

And can someone review the globalid.ecl?

File: thorlcr/graph/thgraph.cpp (hunk 2 above)
Comment (afishbeck) [reply thorlcr/graph/thgraph.cpp]: Yes, good catch, thanks.

**Comment by afishbeck** [comment]
@wangkx updated, please check.

**Comment by wangkx** [comment]
@afishbeck looks good.

**Pull request merged**